File: oldFiles/quickInfo.pl
#!/usr/bin/perl -w

use strict;

my $interactive = "true" if (@ARGV == 0);

print("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n");
print("<disc>\n");
print(STDERR "Enter artist: ") if ($interactive);

my $artist = <>;
chomp($artist);
print("<artist>$artist</artist>\n");
print(STDERR "Enter album: ") if ($interactive);

my $album = <>;
chomp($album);
print("<album>$album</album>\n");
print(STDERR "Enter genre: ") if ($interactive);

my $genre = <>;
chomp($genre);
print("<genre>$genre</genre>\n");
print(STDERR "Enter date: ") if ($interactive);

my $date = <>;
chomp($date);
print("<date>$date</date>\n");
print(STDERR "Enter discnumber (Leave blank if only one): ") if ($interactive);

my $disc = <>;
chomp($disc);
if ($disc ne "") {
   print("<discnumber>$disc</discnumber>\n");
   $disc = <> if (! $interactive);
}

print(STDERR "Enter each track name and press return (Ctrl-D when done)\n")
   if ($interactive);
my $i = 0;
while (<>) {
   chomp();
   print("\n<track>\n");
   print("<title>$_</title>\n");
   print("<tracknumber>", ++$i, "</tracknumber>\n");
   print("</track>\n");
}
print("</disc>\n");

File: scripts/quickInfo.py
#!/usr/bin/python3

import sys
from xml.dom import minidom

xmlDoc = minidom.getDOMImplementation().createDocument(None, 'disc', None)
rootNode = xmlDoc.documentElement

for tagName in ['artist', 'album', 'genre', 'date', 'discnumber']:
    print('Enter', tagName + ': ', file=sys.stderr)
    tagValue = sys.stdin.readline().strip()
    if tagValue != "":
        newNode = rootNode.appendChild(xmlDoc.createElement(tagName))
        newNode.appendChild(xmlDoc.createTextNode(tagValue))

print('Enter each track name and press return (Ctrl-D when done)',
        file=sys.stderr)
tracks = sys.stdin.readlines()
for i in range(len(tracks)):
    trackTag = rootNode.appendChild(xmlDoc.createElement('track'))
    titleTag = trackTag.appendChild(xmlDoc.createElement('title'))
    trackNumTag = trackTag.appendChild(xmlDoc.createElement('tracknumber'))
    trackNumTag.appendChild(xmlDoc.createTextNode(str(i + 1)))
    titleTag.appendChild(xmlDoc.createTextNode(tracks[i].strip()))

xmlDoc.writexml(sys.stdout, addindent='    ', newl='\n', encoding='utf-8')
